File: ProblemSolving/450DSA/JavaScript/src/string/RomanToInteger.js
/**
 * @author Anirudh Sharma
 * 
 * Given a string in roman no format(s) your task is to convert it to an integer.
 * Various symbols and their values are given below.
 */
const romanToInteger = (s) => {
    // Map to store romans numerals
    const romanMap = new Map();
    // Fill the map
    romanMap.set('I', 1);
    romanMap.set('V', 5);
    romanMap.set('X', 10);
    romanMap.set('L', 50);
    romanMap.set('C', 100);
    romanMap.set('D', 500);
    romanMap.set('M', 1000);
    // Length of the given string
    const n = s.length;
    // Variable to store result
    let decimal = romanMap.get(s.charAt(n - 1));
    // Loop for each character from right to left
    for (let i = n - 2; i >= 0; i--) {
        // Check if the character at right of current character is
        // bigger or smaller
        if (romanMap.get(s.charAt(i)) >= romanMap.get(s.charAt(i + 1))) {
            decimal += romanMap.get(s.charAt(i));
        } else {
            decimal -= romanMap.get(s.charAt(i));
        }
    }
    return decimal;
};

const main = () => {
    console.log(romanToInteger("V"));
    console.log(romanToInteger("III"));
};

main();

File: ProblemSolving/450DSA/Python/src/string/RomanToInteger.py
"""
@author Anirudh Sharma

Given a string in roman no format(s) your task is to convert it to an integer.
Various symbols and their values are given below.
"""


def romanToInteger(s):
    # Dictionary of roman numerals
    roman_map = {'I': 1, 'V': 5, 'X': 10,
                 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
    # Length of the given string
    n = len(s)
    # This variable will store result
    decimal = roman_map[s[n - 1]]
    # Loop for each character from right to left
    for i in range(n - 2, -1, -1):
        # Check if the character at right of current character is bigger or smaller
        if roman_map[s[i]] >= roman_map[s[i + 1]]:
            decimal += roman_map[s[i]]
        else:
            decimal -= roman_map[s[i]]
    return decimal


if __name__ == "__main__":
    print(romanToInteger('V'))
    print(romanToInteger('III'))
